mesonbuild/utils/universal.py

```diff
@@ -1415,8 +1415,9 @@ def typeslistify(item: 'T.Union[_T, T.List[_T]]',
         item = T.cast('T.List[_T]', [item])
     if not isinstance(item, list):
         raise MesonException(f'must be an either {types!r}, or an array thereof, not {type(item)!r}')
+
     for i in item:
-        if i is not None and not isinstance(i, types):
+        if not isinstance(i, types):
```

Comment:
The test failures happen because we once do `stringlistify(kwargs.get('install_tag', [None]))` and that fails when we typeslistify None as permitted by types=str.

It's some weird code and clearly the obvious way to do the intent here is `typeslistify(..., [None], (str, type(None)))`.